AVPlayerExample/AudioDevices/ExampleAVPlayerProcessingTap.m

```diff
@@ -0,0 +1,395 @@
+                                  CMItemCount numberFrames,
+                                  MTAudioProcessingTapFlags flags,
+                                  AudioBufferList *bufferListInOut,
+                                  CMItemCount *numberFramesOut,
+                                  MTAudioProcessingTapFlags *flagsOut) {
+    ExampleAVPlayerAudioTapContext *context = (ExampleAVPlayerAudioTapContext *)MTAudioProcessingTapGetStorage(tap);
+    CMTimeRange sourceRange;
+    OSStatus status = MTAudioProcessingTapGetSourceAudio(tap,
+                                                         numberFrames,
+                                                         bufferListInOut,
+                                                         flagsOut,
+                                                         &sourceRange,
+                                                         numberFramesOut);
+    if (status != noErr) {
+        // TODO: It might be useful to fill zeros here.
+        return;
+    } else if(CMTIMERANGE_IS_EMPTY(sourceRange) ||
+              CMTIMERANGE_IS_INVALID(sourceRange)) {
+        return;
+    }
+
+    UInt32 framesToCopy = (UInt32)*numberFramesOut;
+
+    // Produce renderer buffers. These are interleaved, signed integer frames in the source's sample rate.
+    TPCircularBuffer *renderingBuffer = context->renderingBuffer;
+    AVPlayerAudioTapProduceConvertedFrames(renderingBuffer,
+                                           context->renderFormatConverter,
+                                           bufferListInOut,
+                                           framesToCopy,
+                                           &sourceRange,
+                                           kPreferredNumberOfChannels);
+
+    // Produce capturer buffers. We will perform a sample rate conversion if needed.
+    TPCircularBuffer *capturingBuffer = context->capturingBuffer;
+    if (context->capturingSampleRateConversion) {
+        AVPlayerAudioTapProduceFilledFrames(capturingBuffer,
+                                            context->captureFormatConverter,
+                                            context->captureFormatConvertIsPrimed,
+                                            bufferListInOut, context->sourceCache,
+                                            &context->sourceCacheFrames,
+                                            framesToCopy,
+                                            kPreferredNumberOfChannels);
+        context->captureFormatConvertIsPrimed = YES;
+    } else {
+        AVPlayerAudioTapProduceConvertedFrames(capturingBuffer,
+                                               context->captureFormatConverter,
+                                               bufferListInOut,
+                                               framesToCopy,
+                                               &sourceRange,
+                                               kPreferredNumberOfChannels);
+    }
```

Comment:
Lets make sure we are not turning on the capturing sample rate conversion stream on the tap.